### PR TITLE
Features/rings ext

### PR DESCRIPTION
**Pull request by RyanKung**

### CODE REVIEW

File: node/src/backend/extension.rs [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+//!  (import "message_abi" "message_type" (func $message_type (type $ty_message_type)))
+//!   ;; fn handler(param: ExternRef) -> ExternRef
+//!  (func $handler  (param externref) (result externref)
+//!      (return (local.get 0))
+//!  )
+//!  (export "handler" (func $handler))
+//! )
+//!
+//! You can see that this wat/wasm extension defines a handler function and
+//! imported the message_type ABI.
```
Comment (Ma233) [inline node/src/backend/extension.rs:34]:
```suggestion
//! imports the message_type ABI.
```

File: node/bin/rings.rs [hunk 2]
```diff
@@ -426,7 +426,7 @@ async fn daemon_run(args: RunCommand) -> anyhow::Result<()> {
     let processor_clone = processor.clone();
 
     let pubkey = Arc::new(key.pubkey());
-    println!("Signature: {}", Processor::generate_signature(&key));
+    println!("Signautre: {}", Processor::generate_signature(&key));
```
Comment (Ma233) [inline node/bin/rings.rs:429]:
```suggestion
    println!("Signature: {}", Processor::generate_signature(&key));
```

File: node/src/backend/extension.rs [hunk 3]
```diff
@@ -0,0 +1,422 @@
+#![warn(missing_docs)]
+//! The "External" module supports a user-defined message handler based on WebAssembly (Wasm).
```
Comment (Ma233) [inline node/src/backend/extension.rs:2]:
```suggestion
//! This module supports a user-defined message handler based on WebAssembly (Wasm).
```

File: node/src/backend/extension.rs [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+                MaybeBackendMessage::extra,
+            );
+
+            imports! {
+            "message_abi" => {
+                        "message_type"  => msg_type,
+                        "extra" => extra,
+                        "data" => data
+            }
+                }
```
Comment (Ma233) [inline node/src/backend/extension.rs:198]: Indentation need to be fixed

**Review by Ma233 — DISMISSED** [review]

File: derive/Cargo.toml [hunk 5]
```diff
@@ -10,7 +10,9 @@ license = "GPL-3.0"
 proc-macro = true
 
 [features]
-default = []
+default = [
```
Comment (r2d2-rs) [inline derive/Cargo.toml:13]: should remove

**Review by r2d2-rs — APPROVED** [review]

**Pull request merged**